framework/daemonset_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 Authors of spidernet-io
+// SPDX-License-Identifier: Apache-2.0
+package framework_test
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	e2e "github.com/spidernet-io/e2eframework/framework"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func generateExampleDaemonSetYaml(dsName, namespace string) *appsv1.DaemonSet {
+	Expect(dsName).NotTo(BeEmpty())
+	Expect(namespace).NotTo(BeEmpty())
+
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      dsName,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": dsName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": dsName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "samplepod",
+							Image:           "alpine",
+							ImagePullPolicy: "IfNotPresent",
+							Command:         []string{"/bin/ash", "-c", "trap : TERM INT; sleep infinity & wait"},
+						},
+					},
+				},
+			},
+		},
+		Status: appsv1.DaemonSetStatus{
```

Comment:
// the fake clientset will not schedule daemonset replicaset， so mock the number